src/Data/Nat/Coprimality.agda

```diff
@@ -139,11 +138,14 @@ coprime-factors c (divides q₁ eq₁ , divides q₂ eq₂) with coprime-Bézout
 ------------------------------------------------------------------------
 -- Primality implies coprimality.
 
+prime⇒coprime≢0 : ∀ {p} → Prime p →
+                 ∀ {n} .{{_ : NonZero n}} → n < p → Coprime p n
+prime⇒coprime≢0 {p} pr@(prime _) n<p {d} (d∣p , d∣n)
+  with prime⇒irreducible pr d∣p
+... | inj₁ d≡1 = d≡1
+... | inj₂ refl with () ← ≤⇒≯ (∣⇒≤ d∣n) n<p
+
 prime⇒coprime : ∀ m → Prime m →
                 ∀ n → 0 < n → n < m → Coprime m n
-prime⇒coprime (suc (suc _)) p _ _ _ {0} (0∣m , _) =
-  contradiction (0∣⇒≡0 0∣m) λ()
-prime⇒coprime (suc (suc _)) _ _ _ _ {1} _         = refl
-prime⇒coprime (suc (suc _)) p (suc _) _ n<m {(suc (suc _))} (d∣m , d∣n) =
-  contradiction d∣m (p 2≤d d<m)
-  where 2≤d = s≤s (s≤s z≤n); d<m = <-≤-trans (s≤s (∣⇒≤ d∣n)) n<m
+prime⇒coprime _ p _ z<s = prime⇒coprime≢0 p
```

Comment:
Now, there's a possible deprecation opportunity?

Reply:
Elsewhere in the library (e.g. divisibility proofs) I've just taken advantage of v2.0 to change the type of the proof entirely when introducing instance arguments. So the new proof above would assume the name `prime=>comprime`. You would need to add an entry in the very bottom of the CHANGELOG where we list all proofs that have been changed in such a way.